hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/CommonUtils.java

```diff
@@ -88,4 +135,143 @@ public static boolean productWouldOverflow(final long multiplier, final long mul
     static void setSha384HashTag(final String sha384HashTag) {
         CommonUtils.sha384HashTag = sha384HashTag;
     }
+
+    public static HederaFunctionality functionOf(final TransactionBody txn) throws UnknownHederaFunctionality {
+        if (txn.hasContractCall()) {
```

Comment:
This should be replaced with a switch statement IMO. Can be done in a follow-up PR.

Reply:
Each if statement has different .hasXXX calls, I would like to see how you use switch statement IMO. Any code example?

Reply:
Right, you cannot use these methods. But you can do a switch based on `TransactionBody.data().kind()`. You can find an example in `TransactionDispatcher`.

Reply:
I replaced if with switch. Thank you @netopyr for the good suggestion!